inference-engine/src/transformations/src/transformations/common_optimizations/convert_nms_gather_path_to_unsigned.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/convert_nms_gather_path_to_unsigned.hpp"
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/op/util/broadcast_base.hpp>
+#include <ngraph/op/util/gather_base.hpp>
+#include <ngraph/rt_info.hpp>
+#include <memory>
+#include "itt.hpp"
+#include "ngraph/node.hpp"
+
+using namespace ngraph;
+using namespace std;
+
+namespace ngraph {
+namespace pass {
+
+class TRANSFORMATIONS_API InitNMSPath;
+class TRANSFORMATIONS_API PropagateNMSPath;
+class TRANSFORMATIONS_API UpdateConvertGather;
+
+}  // namespace pass
+}  // namespace ngraph
+
+class ngraph::pass::InitNMSPath: public ngraph::pass::MatcherPass {
+public:
+    NGRAPH_RTTI_DECLARATION;
+
+    InitNMSPath() {
+        MATCHER_SCOPE(InitNMSPath);
+
+        auto nms_pattern = pattern::wrap_type<opset8::NonMaxSuppression>();
+
+        ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher &m) {
+            auto nms_node = dynamic_pointer_cast<opset8::NonMaxSuppression>(m.get_match_root());
+            bool res = false;
+            if (!nms_node)
+                return false;
+
+            const auto& out_nodes = nms_node->output(0).get_target_inputs();
+            for (const auto& out_node : out_nodes) {
+                auto& out_rt_info = out_node.get_node()->get_rt_info();
+                out_rt_info["NMS_SELECTED_INDICES"] = std::make_shared<ngraph::VariantWrapper<string>>("");
+                res = true;
+            }
+            return res;
```

Comment:
Suggest to mark only output port for NMS. And for other operations you will check inputs.
```suggestion
            auto & rt_info = m.get_match_root()->output(0);
            out_rt_info["NMS_SELECTED_INDICES"] = std::make_shared<ngraph::VariantWrapper<string>>("");
            return true;
```